src/diagapp.jl

```diff
@@ -1,25 +1,37 @@
 # Diagonal Approximation for trace of inverse of matrix
+# Values from Extrapolation Methods for Estimating Trace of the Matrix Inverse by P. Fika
 
-export diagapp
+export diagapp, diagapp!
 
 using LinearAlgebra
 
+struct diagappspace
+    A::AbstractArray{<:Any, 2}
```

Comment:
Use type parameters here.

Reply:
Sorry, can you give a quick example? (Will correct it for the rest of the errors as well)